### PR TITLE
fix relation

### PR DESCRIPTION
**Pull request by fbielejec**
fixes for the withdraw relation: 

- it could not represent something that is larger that P in the field
- added a test with some representative values
- use trait to retrieve public inputs

### CODE REVIEW

File: blender-cli/src/withdraw.rs [hunk 1]
```diff
@@ -49,8 +49,8 @@ pub(super) fn do_withdraw(
         None => account_from_keypair(&keypair_from_string(&app_state.caller_seed)),
         Some(recipient) => recipient,
     };
-
-    let recipient_bytes = [0u8; 32]; // its not checked anyway
+    let recipient_bytes: [u8; 32] = recipient.clone().into();
+    println!("recipient_bytes: {:?}", recipient_bytes);
```
Comment (deuszx) [inline blender-cli/src/withdraw.rs:53]: leftover

**Comment by deuszx** [comment]
Not sure where's the fix 🤔

**Comment by fbielejec** [comment]
> Not sure where's the fix thinking

https://github.com/Cardinal-Cryptography/HouseSnark/pull/31/files#diff-a6022c31aeadb8d5fd346befddf6e9843d7c324e8b1139ed58a1357e716222b2R140

File: house-snark/src/relations/blender/withdraw.rs [hunk 2]
```diff
@@ -355,17 +358,9 @@ mod tests {
...
-        let input = [
-            circuit.fee.clone(),
-            circuit.recipient.clone(),
-            circuit.token_id.clone(),
-            circuit.old_nullifier.clone(),
-            circuit.new_note.clone(),
-            circuit.token_amount_out.clone(),
-            circuit.merkle_root.clone(),
-        ];
+        let input = circuit.public_input();
```
Comment (deuszx) [inline house-snark/src/relations/blender/withdraw.rs:361]: So there's a trait that is implemented for an array of elements?

Also, why not 
```rust
        let input = circuit.public_input().try_into().unwrap())
```
right away?

**Review by deuszx — APPROVED** [review]

File: house-snark/src/relations/blender/withdraw.rs (hunk 2 above)
Comment (fbielejec) [reply house-snark/src/relations/blender/withdraw.rs:361]: what would be the advantage, because AFAIK the two are equivalent?

**Review by Yithis — APPROVED** [review]

**Pull request merged**